src/drawers/drawer.ts

```diff
@@ -36,16 +36,20 @@ export module Drawers {
      * Constructs a Drawer
      *
      * @constructor
-     * @param{string} key The key associated with this Drawer
+     * @param {Dataset} dataset The dataset associated with this Drawer
      */
-    constructor(key: string) {
-        this.key = key;
+    constructor(dataset: Dataset) {
+        this._dataset = dataset;
     }
 
     public setup(area: D3.Selection) {
       this._renderArea = area;
     }
 
+    protected dataset() {
```

Comment:
There was a comment that @aicioara made that at his previous workplace, there was actually no concept of `protected` instance variables and that the way to get at the values would be through `protected` getters/setters.  After hearing about this, I think I am a huge fan of this model.

The biggest thing for me personally is that this will guard against setting variables in a parent class when they should have been only set at initialization.

Another benefit is that this will guard against API breaks since it will essentially allow us to put architecture surround these instance variables instead of setting them directly / indirectly